sources/api/migration/migrations/v1.0.6/add-shibaken/src/main.rs

```diff
@@ -0,0 +1,23 @@
+#![deny(rust_2018_idioms)]
+
+use migration_helpers::common_migrations::{AddMetadataMigration, SettingsMetadata};
+use migration_helpers::{migrate, Result};
+use std::process;
+
+/// We added a new setting metadata, `host-containers.admin.user-data.setting-generator`
+fn run() -> Result<()> {
+    migrate(AddMetadataMigration(vec![SettingsMetadata {
+        metadata: &["setting-generator"],
+        setting: "settings.host-containers.admin.user-data",
+    }]))
```

Comment:
@tjkirch - if we may eventually need a `ReplaceMetadataMigration` to complement this one, would it be better to anticipate that with an interface like this?
```
    migrate(ReplaceMetadataMigration(vec![SettingsMetadata {
        metadata: &["setting-generator"],
        new_setting: Some("settings.host-containers.admin.user-data"),
        old_setting: None.
    }]))
```

Having a family of `AddMetadataMigration`, `ReplaceMetadataMigration`, `RemoveMetadataMigration` isn't the worst thing - it has the benefit of being explicit - but it does mean we'd need a second migration binary whenever we're adding as well as replacing.